File: student/src/main/java/com/ktxdevelopment/model/entity/Faculty.java
package com.ktxdevelopment.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import org.springframework.context.annotation.Lazy

import java.util.List;

@Entity(name = "faculty")
@Table(name = "faculties")
class Faculty {

    @Id
    @GeneratedValue
    private Integer id;

    private String name;

    @Lazy
    @OneToMany
    List<Course> courses;
}

File: student/src/main/java/com/ktxdevelopment/model/entity/Faculty.kt
package com.ktxdevelopment.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.springframework.context.annotation.Lazy

@Entity
@Table(name = "faculty")
data class Faculty(
    @Id
    @GeneratedValue
    val id: Int,
    val name: String,
    @Lazy val courses: List<Course>
)